File: Backend/carShop/src/main/java/com/contrader/carShop/dto/AcquistiDTO.java
package com.contrader.carShop.dto;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import com.contrader.carShop.model.Auto;
import com.contrader.carShop.model.User;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;


@Data
@AllArgsConstructor
@NoArgsConstructor
@JsonIdentityInfo(generator= ObjectIdGenerators.PropertyGenerator.class, property="id")
public class AcquistiDTO {

    private Long id;

    private Long idOrdine;
    private int quantita;

    private double prezzo;
    private boolean acquistato;

    private String dataAcquisto;
    private String orarioAcquisto;

    private User user;

    private Auto auto;

}

File: Frontend/src/dto/acquistidto.ts
import { AutoDTO } from "./auto";
import { UserDTO } from "./userdto";

export class AcquistiDTO {

    id: number;
    idOrdine: number;
    quantita: number;
    prezzo: number;
    acquistato: boolean;
    dataAcquisto: string;
    orarioAcquisto: string;
    user: UserDTO;
    auto: AutoDTO;

    constructor(
                id: number,
                idOrdine: number,
                quantita: number,
                prezzo: number,
                isAcquistato: boolean,
                dataAcquisto: string,
                orarioAcquisto: string,
                user: UserDTO,
                auto: AutoDTO){
        this.id = id;
        this.idOrdine = idOrdine;
        this.quantita = quantita;
        this.prezzo = prezzo;
        this.acquistato = isAcquistato;
        this.dataAcquisto = dataAcquisto;
        this.orarioAcquisto = orarioAcquisto;
        this.user = user;
        this.auto = auto;
    }
}